alibi/explainers/similarity/backends/pytorch/base.py

```diff
@@ -50,8 +50,27 @@ def get_grads(
         loss = loss_fn(output, Y)
         loss.backward()
         model.train(initial_model_state)
-        return np.concatenate([_PytorchBackend.to_numpy(param.grad).reshape(-1)  # type: ignore [arg-type] # see #810
-                               for param in model.parameters()])
+
+        return np.concatenate([_PytorchBackend._grad_to_numpy(grad=param.grad, name=name)
+                               for name, param in model.named_parameters()
+                               if param.grad is not None])
```

Comment:
Note that torch tensors have a `grad` attribute by default set to `None` so we don't have to check it exist first!